src/resolution_functions/instrument.py

```diff
@@ -449,6 +446,33 @@ def _get_model_data(self, model_name: Optional[str] = None, **kwargs) -> tuple[M
                                        **ChainMap(*configurations, model['parameters']))
         return model, model_name
 
+    def _resolve_model(self, model_name: str) -> dict:
+        """
+        Returns the data for the `model_name` model, taking into account aliases.
+
+        Parameters
+        ----------
+        model_name
+            The name of the model whose data to retrieve.
+
+        Returns
+        -------
+        model_data
+            The dictionary with the model's data.
+        """
+        try:
+            model = self._models[model_name]
+        except KeyError:
+            raise InvalidModelError(model_name, self)
+
+        if isinstance(model, str):
+            try:
+                model = self._models[model]
+            except KeyError:
+                raise InvalidModelError(model, self)
```

Comment:
Could/should this recursively call `_resolve_model`? Would that be clearer?

Reply:
We have decided to enforce in the YAML data file spec that, if the model is a string, it *must* point to a model that has actual data, i.e. forbidding this kind of chaining.

Reply:
This eliminates some difficult (e.g. circular) cases at the acceptable cost of sometimes writing a few more characters in YAML.